tools/make_patch.c

```diff
@@ -0,0 +1,454 @@
+		if (fseek(orig_rom, current_offset, SEEK_SET)) {
+			error_exit("Error: Cannot seek to \"vc_patch %s\" in the original ROM\n", current_hook->name);
+		}
+		if (fseek(new_rom, current_offset, SEEK_SET)) {
+			error_exit("Error: Cannot seek to \"vc_patch %s\" in the new ROM\n", current_hook->name);
+		}
+		int length;
+		if (argc == 2) {
+			length = parse_number(argv[1], 0);
+		} else {
+			const struct Symbol *current_hook_end = symbol_find_cat(symbols, current_hook->name, "_End");
+			length = current_hook_end->offset - current_offset;
+		}
+		buffer_append(patches, &(struct Patch){current_offset, length});
+		bool modified = false;
+		if (length == 1) {
+			int c = getc(new_rom);
+			modified = c != getc(orig_rom);
+			fprintf(output, isupper((unsigned)command[0]) ? "0x%02X" : "0x%02x", c);
+		} else {
+			if (command[strlen(command) - 1] != '/') {fprintf(output, command[strlen(command) - 1] == '_' ? "a%d: " : "a%d:", length);}
+			for (int i = 0; i < length; i++) {
+				if (i) {
+					putc(' ', output);
+				}
+				int c = getc(new_rom);
+				modified |= c != getc(orig_rom);
+				fprintf(output, isupper((unsigned)command[0]) ? "%02X" : "%02x", c);
+			}
+		}
+		if (!modified) {
+			fprintf(stderr, PROGRAM_NAME ": Warning: \"vc_patch %s\" doesn't alter the ROM\n", current_hook->name);
+		}
+
+	} else if (!strcmp(command, "dws") || !strcmp(command, "DWS") || !strcmp(command, "dws_") || !strcmp(command, "DWS_") || !strcmp(command, "dws/") || !strcmp(command, "DWS/")) {
+		if (argc < 1) {
+			error_exit("Error: Invalid arguments for command: \"%s\"\n", command);
+		}
+		if (command[strlen(command) - 1] != '/') {fprintf(output, command[strlen(command) - 1] == '_' ? "a%d: " : "a%d:", argc * 2);}
+		for (int i = 0; i < argc; i++) {
+			int value = parse_arg_value(argv[i], false, symbols, current_hook->name);
+			if (value > 0xffff) {
+				error_exit("Error: Invalid value for \"%s\" argument: 0x%x\n", command, value);
+			}
+			if (i) {
+				putc(' ', output);
+			}
+			fprintf(output, isupper((unsigned)command[0]) ? "%02X %02X": "%02x %02x", value & 0xff, value >> 8);
+		}
+
+	} else if (!strcmp(command, "db") || !strcmp(command, "DB") || !strcmp(command, "db_") || !strcmp(command, "DB_") || !strcmp(command, "db/") || !strcmp(command, "DB/")) {
+		if (argc != 1) {
+			error_exit("Error: Invalid arguments for command: \"%s\"\n", command);
+		}
+		int value = parse_arg_value(argv[0], false, symbols, current_hook->name);
+		if (value > 0xff) {
+			error_exit("Error: Invalid value for \"%s\" argument: 0x%x\n", command, value);
+		}
+		if (command[strlen(command) - 1] != '/') {fputs(command[strlen(command) - 1] == '_' ? "a1: " : "a1:", output);}
+		fprintf(output, isupper((unsigned)command[0]) ? "%02X" : "%02x", value);
+
+	} else if (!strcmp(command, "hex") || !strcmp(command, "HEX") || !strcmp(command, "HEx") || !strcmp(command, "Hex") || !strcmp(command, "heX") || !strcmp(command, "hEX")) {
```

Comment:
No support for `hex~` and `HEX~` yet; that's fine, after this merges we can port the updated support for `/` to gold+crystal, then PR crystal to document and support and use `hex~`, then port that back to red+yellow.

Reply:
`hex~` is now supported, and will need porting to gold+crystal for use with `hJoyPressed`.

Reply:
Also documenting `hex~`, and the `/` suffix for value series commands, in pokecrystal.